lib/src/builder/builder.dart

```diff
@@ -75,18 +86,16 @@ class OverReactBuilder extends Builder {
     }
 
     if (outputs.isNotEmpty) {
-      final outputId = buildStep.inputId.changeExtension(outputExtension);
-
       // Verify that the library file has an `.over_react.g.dart` part.
-      final expectedPart = p.basename(outputId.path);
-      final partUris = libraryUnit.directives
-        .whereType<PartDirective>()
-        .map((p) => p.uri.stringValue);
-      if (!partUris.contains(expectedPart)) {
+      if (!hasOutputPartDirective()) {
         log.warning('Missing "part \'$expectedPart\';".');
       }
 
       await _writePart(buildStep, outputId, outputs);
+    } else {
+      if (hasOutputPartDirective()) {
+        log.warning('An over react part directive was found in ${buildStep.inputId.path}, but no code was generated.');
```

Comment:
It might be good to indicate potential causes of this so that the user can attempt to correct them. (E.g., "The part directive may be unnecessary if the file does not contain any concrete components.")